crates/re_chunk_store/src/query.rs

```diff
@@ -0,0 +1,236 @@
+    ) -> Option<TimeInt> {
+        let temporal_chunk_ids_per_timeline =
+            self.temporal_chunk_ids_per_entity.get(entity_path)?;
+        let temporal_chunk_ids_per_component = temporal_chunk_ids_per_timeline.get(timeline)?;
+
+        let mut time_min = TimeInt::MAX;
+        for temporal_chunk_ids_per_time in temporal_chunk_ids_per_component.values() {
+            let Some(time) = temporal_chunk_ids_per_time
+                .per_start_time
+                .first_key_value()
+                .map(|(time, _)| *time)
+            else {
+                continue;
+            };
+            time_min = TimeInt::min(time_min, time);
+        }
+
+        (time_min != TimeInt::MAX).then_some(time_min)
+    }
+
+    /// Returns the most-relevant chunk(s) for the given [`LatestAtQuery`].
+    ///
+    /// The [`ChunkStore`] always work at the [`Chunk`] level (as opposed to the row level): it is
+    /// oblivious to the data therein.
+    /// For that reason, and because [`Chunk`]s are allowed to temporally overlap, it is possible
+    /// that a query has more than one relevant chunk.
+    ///
+    /// The caller should filter the returned chunks further (see [`Chunk::latest_at`]) in order to
+    /// determine what exact row contains the final result.
+    ///
+    /// If the entity has static component data associated with it, it will unconditionally
+    /// override any temporal component data.
+    pub fn latest_at(
+        &self,
+        query: &LatestAtQuery,
+        entity_path: &EntityPath,
+        component_name: ComponentName,
+    ) -> Vec<Arc<Chunk>> {
+        re_tracing::profile_function!(format!("{query:?}"));
+
+        self.query_id.fetch_add(1, Ordering::Relaxed);
+
+        // Reminder: if a chunk has been indexed for a given component, then it must contain at
+        // least one non-null value for that column.
+
+        if let Some(static_chunk) = self
+            .static_chunk_ids_per_entity
+            .get(entity_path)
+            .and_then(|static_chunks_per_component| {
+                static_chunks_per_component.get(&component_name)
+            })
+            .and_then(|chunk_id| self.chunks_per_chunk_id.get(chunk_id))
+        {
+            return vec![Arc::clone(static_chunk)];
+        }
+
+        if let Some(temporal_chunk_ids) = self
+            .temporal_chunk_ids_per_entity
+            .get(entity_path)
+            .and_then(|temporal_chunk_ids_per_timeline| {
+                temporal_chunk_ids_per_timeline.get(&query.timeline())
+            })
+            .and_then(|temporal_chunk_ids_per_component| {
+                temporal_chunk_ids_per_component.get(&component_name)
+            })
+            .and_then(|temporal_chunk_ids_per_time| {
+                temporal_chunk_ids_per_time
+                    .per_start_time
+                    .range(..=query.at())
+                    .next_back()
+                    .map(|(_time, chunk_ids)| chunk_ids)
```

Comment:
This looks very obviously wrong for cases where a chunk completely overlaps another one? write a test and fix this code.